lib/hlclock.ex

```diff
@@ -48,6 +48,46 @@ defmodule HLClock do
   end
 
   @doc """
+  Functionally equivalent to using send_timestamp. This generates a timestamp
+  for local causality tracking.
+  """
+  def now do
+    GenServer.call(HLClock.Server, :send_timestamp)
+  end
+
+  @doc """
+
+  Create a millisecond granularity DateTime struct representing the logical time
+  portion of the Timestamp.
+
+  Given that this representation loses the logical counter and node information,
+  it should be used as a reference only. Including the counter in the DateTime
+  struct would create absurd but still ordered timestamps.
+
+  ## Example
+
+      iex> {:ok, t0} = Timestamp.new(1410652800000, 0, 0)
+      {:ok, %HLClock.Timestamp{counter: 0, node_id: 0, time: 1410652800000}}
+      iex> encoded = Timestamp.encode(t0)
+      <<1, 72, 113, 117, 132, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>
+      iex> << time_and_counter :: size(64), _ :: size(64) >> = encoded
+      iex> DateTime.from_unix(time_and_counter, :microsecond)
+      {:ok, #DateTime<4899-07-30 06:31:40.800000Z>}
+  """
+  def to_datetime(%Timestamp{time: t}) do
+    with {:ok, dt} <- DateTime.from_unix(t, :millisecond) do
```

Comment:
does this actually do what we want? We aren't going to combine the time and counter portions here?

Reply:
Yeah, I think it does. Otherwise what you're saying is you want timestamps that are inscrutable. I mean unless that's what you want.

Reply:
I thought we figured out how to do this with both the time and the counter being packed together?

Reply:
I did not. The issue is pretty simple. For small counters (usually the case), you could probably come up with something where it works, but using 64 bits for time isn't because `DateTime` can actually represent the full range of 64 bit values.

Max time in milliseconds (i.e. `t_max = 253402300799999999`) is actually on the order of 57 bits, so treating it as a 64 bit value is prone to issues. `DateTime` will throw when our value is greater than `t_max`.

Does that make sense?